File: 0045-jump-game-ii/0045-jump-game-ii.go
func jump(nums []int) int {
    for i:= 1; i < len(nums); i++{
        if nums[i - 1] > nums[i] + i {
            nums[i] =  nums[i-1]
        } else {
            nums[i] =  nums[i] + i
        }
    }
    
    result := 0
    position := 0
    for position < len(nums) - 1 {
        result++
        position = nums[position]
    }
    
    return result
}

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        for(int i = 1; i < nums.length; i++){
            nums[i] = Math.max(nums[i - 1], nums[i] + i);
        }
        
        int result = 0;
        int position = 0;
        
        while(position < nums.length - 1){
            result++;
            position = nums[position];
        }
        
        return result;
    }
}